action/lib/Database.js

```diff
@@ -3,207 +3,61 @@
 
 const common = require('./common');
 
-// constructor for DB object - a thin, promise-loving wrapper around nano
-module.exports = function(dbURL, dbName) {
-    var nano = require('nano')(dbURL);
-    this.db = nano.db.use(dbName);
-    var utilsDB = this;
+module.exports = function() {
 
-    this.getWorkerID = function(availabeWorkers) {
-
-        return new Promise((resolve, reject) => {
-            var workerID = availabeWorkers[0] || 'worker0';
-
-            if (availabeWorkers.length > 1) {
-                utilsDB.db.view('triggerViews', 'triggers_by_worker', {reduce: true, group: true}, function (err, body) {
-                    if (!err) {
-                        var triggersByWorker = {};
+    var database = this;
 
-                        availabeWorkers.forEach(worker => {
-                            triggersByWorker[worker] = 0;
-                        });
+    // init for DB object - a thin, promise-loving wrapper around nano / documentdb
+    this.initDB = function(config) {
 
-                        body.rows.forEach(row => {
-                            if (row.key in triggersByWorker) {
-                                triggersByWorker[row.key] = row.value;
-                            }
-                        });
-
-                        // find which worker has the least number of assigned triggers
-                        for (var worker in triggersByWorker) {
-                            if (triggersByWorker[worker] < triggersByWorker[workerID]) {
-                                workerID = worker;
-                            }
-                        }
-                        resolve(workerID);
-                    } else {
-                        reject(err);
-                    }
-                });
+        config.type = typeof config.type  !== 'undefined' ?  config.type  : "couchdb";
+        var db = {};
+        return new Promise((resolve, reject) => {
+            if(config.type === "couchdb") {
+                console.log("using couchdb");
+                var couchdb = require('./couchdb');
+                db = new couchdb(config.dburl, config.dbname);
+                database.utilsDB = db;
+                resolve();
             }
-            else {
-                resolve(workerID);
+            else if(config.type === "cosmosdb") {
+                console.log("using cosmosdb");
+                var cosmosdb = require('./cosmosdb');
+                db = new cosmosdb(config.dburl, config.masterkey);
+                db.init(config.rootdb, config.dbname)
+                    .then((res) => {
+                        database.utilsDB = db;
+                        resolve();
+                    })
+                    .catch((err) => {
+                        reject(err);});
             }
+            else
+                reject("No db type to initialize");
```

Comment:
I think `type must be couchdb or cosmosdb; found $config.type` would be more clear?

Reply:
Will make this change